datanode/sqlstore/accounts.go

```diff
@@ -33,14 +33,16 @@ var accountOrdering = TableOrdering{
 
 type Accounts struct {
 	*ConnectionSource
-	cache     map[accountCacheKey]entities.Account
-	cacheLock sync.Mutex
+	cache            map[accountCacheKey]entities.Account
+	idToAccountCache map[entities.AccountID]entities.Account
+	cacheLock        sync.Mutex
```

Comment:
Should this be an RWMutex?

Reply:
apparently RWMutex does not let you upgrade the read lock to a write lock (bit surprising), you have to explicity release the RLock and then obtain the Lock, and calling RUnlock when you don't have the RLock is a runtime error(?!).  So you can't then rely on defer to unlock.  Which means the code becomes a bit tangled, I don't think the perf benefit is worth it here.  Maybe I've missed something?

Reply:
Well just Lock when you write and Rlock in any other case. I could imagine the perf benefits could be there seeing that you implemented that specifically because there was lots of calls to the accounts, so the rlock would allow to read that map conccurrently.

Reply:
yeah, because its just a map lookup so contention would be very low (writes are v.rare) I was 50/50 on it tbh given the additional logic needed,  I'll make the change

Reply:
ok, thats done